README.md

```diff
@@ -1,2 +1,178 @@
+              <name>listener</name>
+              <value>com.epam.reportportal.testng.ReportPortalTestNGListener</value>
+           </property>
+           ```
+        2. **With overloading properties**
+           ```
+           <property>
+              <name>listener</name>
+              <value>com.znsio.rpi.listener.ReportPortalListener</value>
+           </property>
+           ```
+    2. **Approach-3:** Use the following snippet to configure ReportPortal listener when creating TestNG file
+       programmatically
+        1. **Without overloading properties**
+           ```
+           ReportPortalTestNGListener reportPortalTestNGListener = new ReportPortalTestNGListener();
+           testNG.addListener(reportPortalTestNGListener);
+           ```
+        2. **With overloading properties**
+           ```
+           ReportPortalListener reportPortalListener = new ReportPortalListener();
+           testNG.addListener(reportPortalListener);
+           ```
+    3. For more details on listener configuration,
+       refer [reportportal-agent-java-testNG](https://github.com/reportportal/agent-java-testNG)
+2. **Properties file:** Create a `reportportal.properties` file either in `src/main/resources/` or
+   in `src/test/resources` directory in your automation framework with the following mandatory attributes
+   ```
+   rp.endpoint = <Your Endpoint>
+   rp.uuid = <Your UDID>
+   rp.launch = <Your Launch Name>
+   rp.project = <Your Project Name>
+   ```
+3. **ReportPortal Attributes and Properties Overlaoding:** Use the `ReportPortalPropertiesOverloader` class defined
+   under `src/main/java/com/znsio/rpi/properties` path to change any attributes/property value for ReportPortal at
+   runtime
+4. **Launch attributes related to CI (Pipeline) execution**: The `setPipelineAttributes` method
+   of `ReportPortalPropertiesOverloader` class takes care of setting the following attributes in case the test execution
+   is happening on CI (Pipeline)
+   ```
+   BUILD_ID
+   AGENT_NAME
+   BRANCH_NAME
+   ```
+   All these attributes will be set as launch attributes if the execution is happening on CI, and if they are either set at
+   System property or System environment variable. If your pipeline is using different keys to set the above attributes
+   then the ones we're using above, for each such key, you can define the new key value in the `config.proprties` file
+   of your automation framework like below:
+   ```
+   BUILD_ID=BUILD_BUILDID
+   BRANCH_NAME=BUILD_SOURCEBRANCHNAME
+   ```
+5. For more details on configuration and additional parameters for `reportportal.properties`,
+   refer [reportportal-client-java](https://github.com/reportportal/client-java)
+   and [reportportal-agent-java-testNG](https://github.com/reportportal/agent-java-testNG)
+
+# How to use the ReportPortal utilities
+
+1. `ReportPortalPropertiesOverloader` class takes care of dynamically setting Report Portal properties at run time like
```

Comment:
The consumer does not need to do this, right?

Reply:
The details are given merely for understanding on the implementation of how the properties are being overloaded at run time. I've removed the usage of word "Use" from each statement to avoid confusion that this is something consumer can access.

Reply:
Let's not add documentation for internal implementation (same as for ApplitoolsIntegration)

Reply:
Removed documentation for all private methods